### PR TITLE
feat: add MTA build & deploy task providers

### PR DESCRIPTION
**Pull request by sapirpol**
Before submitting a pull request, please ensure the following:

## Description

Please explain the changes you made.

Add a link to the design (if applicable).

### Checklist

- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [X] Relevant logs were added
- [X] Linting run locally successfully
- [X] All tests pass
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: tests/extension.spec.ts [hunk 1]
```diff
@@ -94,19 +113,13 @@ describe("Extension unit tests", () => {
       configSettingsMock.verify();
     });
 
-    it("activate - add subscriptions", async () => {
```
Comment (allacherkes) [inline tests/extension.spec.ts:97]: We dont need this test or is it tested in other place?

File: src/utils/utils.ts [hunk 2]
```diff
@@ -149,4 +150,14 @@ export class Utils {
     const cfHome = get(process, "env.CF_HOME", join(os.homedir(), ".cf"));
     return join(cfHome, "config.json");
   }
+
+  public static async isLoggedInToCF(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (sapirpol) [reply src/utils/utils.ts:154]: yes, we use it in deploy mtar command

File: tests/extension.spec.ts (hunk 1 above)
Comment (sapirpol) [reply tests/extension.spec.ts:97]: I check it here https://github.com/SAP/vscode-mta-tools/blob/move_task_provider/tests/extension.spec.ts#L79

File: .eslintrc.json [hunk 3]
```diff
@@ -22,8 +22,11 @@
     ],
     "@typescript-eslint/strict-boolean-expressions": [
       "error",
-      { "allowString": true, "allowNullableString": true }
-    ]
+      { "allowString": true, "allowNullableString": true, "allowAny": true }
```
[the remark this answers is not in the document]
Comment (sapirpol) [reply .eslintrc.json:25]: we get errors in buildTask and deployTask, same as before

File: src/utils/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (allacherkes) [reply src/utils/utils.ts:154]: We dont use it there

File: .eslintrc.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (allacherkes) [reply .eslintrc.json:25]: we dont have any there, maybe in other place?

**Review by allacherkes — APPROVED** [review]

**Comment by rayakoren** [comment]
@sapirpol please fix coverage

**Review by rayakoren — APPROVED** [review]

**Pull request merged**